src/components/organisms/NavBarSchedule/calendar.ts

```diff
@@ -0,0 +1,42 @@
+import { eventEndTime, eventStartTime } from "utils/event";
+import { PersonalizedVenueEvent } from "types/venues";
+import { VenueEvent } from "types/venues";
+import { WithVenueId } from "utils/id";
+
+import { ics } from "./ics";
+
+export const createCalendar = (
+  allEvents: PersonalizedVenueEvent[] | WithVenueId<VenueEvent>[],
+  filename: string
+) => {
+  const cal = ics();
+  if (!cal) return;
+  allEvents.map((event: PersonalizedVenueEvent | WithVenueId<VenueEvent>) => {
+    return cal.addEvent(
+      event.id!,
+      event.name,
+      event.host || "",
+      window.location.origin.toString() + "/in/" + event.venueId,
```

Comment:
Look in `utils/url` for this sort of thing. Our preference would be to use a pattern with `new URL()`. While it's not in `staging` yet, there is a new function for exactly this purpose in https://github.com/sparkletown/sparkle/pull/1396

- https://github.com/sparkletown/sparkle/pull/1396/files#diff-aab4ceb6d3254a7b948f95d8057cd8d91cd43c09f95569885cbce6d8b079d926
- (see also my review suggestion for renaming it https://github.com/sparkletown/sparkle/pull/1396/files#r635905264)

```typescript
export const getFullVenueInsideUrl = (venueId: string) =>
  new URL(venueInsideUrl(venueId), window.location.origin).href;
```

Reply:
For now I just added this new function in the same file, I hope that's okay.

Reply:
Yup, that works well :) That way when it comes time to merge things together, git should just see it as already existing code/similar, and 'just work' without us having to do anything more :)